### ISSUE TITLE
Trailing comma in enum class causes a random semicolon on its own line

### ISSUE THREAD

**Issue by zuhriyansauqi** [pinterest/ktlint#2954]
## Expected Behavior
With the trailing-comma rule enabled, enum classes with multiple entries and additional functions in their body should be formatted without placing the ; on a separate line.

- The Kotlin [Coding Conventions](https://kotlinlang.org/docs/coding-conventions.html) do allow trailing commas, but it doesn’t suggest placing a semicolon alone on its own line.
- IntelliJ’s default formatting also doesn’t move the semicolon to its own line when trailing commas are enabled.

## Observed Behavior
When KtLint is run with trailing comma support enabled, the formatter places the semicolon after the enum entries on its own line, like this:
```kotlin
enum class MyEnum {
    A,
    B,
    ;

    fun printSomething() {
        println("Hello from MyEnum!")
    }
}
```
Note that if trailing commas are disabled or if there’s no code below the enum entries, the semicolon is not placed on its own line.

## Steps to Reproduce
1. Enable trailing commas in .editorconfig. For example:
```
[*.{kt,kts}]
ij_kotlin_allow_trailing_comma_on_declaration_site = true
ij_kotlin_allow_trailing_comma_on_call_site = true
```

2. Create an enum that has multiple entries and at least one function (or property) in the body:
```kotlin
enum class MyEnum {
    A,
    B;

    fun printSomething() {
        println("Hello from MyEnum!")
    }
}
```

3. Run KtLint with --format.
4. See the semicolon placed on its own line after the last entry.

## Your Environment

* Version of ktlint used: 1.5.0
* Relevant parts of the `.editorconfig` settings
```
[*.{kt,kts}]
ij_kotlin_allow_trailing_comma_on_declaration_site = true
ij_kotlin_allow_trailing_comma_on_call_site = true
```
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): Gradle
* Version of Gradle used (if applicable): 8.10.2
* Operating System and version: MacOS 15.3.2

**Comment by paul-dingemans** [comment]
The [Kotlin coding conventions](https://kotlinlang.org/docs/reference/coding-conventions.html#trailing-commas) encourages the usage of trailing commas on the declaration site, but leaves it to the developer's discretion to use trailing commas on the call site. But next to this, it also states that usage of trailing commas has several benefits:
* It makes version-control diffs cleaner – as all the focus is on the changed value.
* It makes it easy to add and reorder elements – there is no need to add or delete the comma if you manipulate elements.
* It simplifies code generation, for example, for object initializers. The last element can also have a comma.

So ktlint chooses to add a trailing comma to the last element in the enum as well. To get benefits above, the semicolon is moved to a separate line. This is not configurable. Also see https://pinterest.github.io/ktlint/latest/faq/#can-a-new-toggle-be-added-to-optionally-enabledisable-format-code-in-a-particular-way
